src/bisection.ts

```diff
@@ -88,6 +88,17 @@ export class ChallengeManager {
     this.root = generateRoot(stateHashes);
   }
 
+  /**
+   * This only works with full, binary trees. For now, we are padding the leaves with leaves of sha256('0').
+   */
+  public get depth() {
```

Comment:
Following the logic here, this is a function of `this.consensusStep` and `this.disputedStep`?

Reply:
`depth` is a function of `this.consensusStep`, `this.disputedStep`, and `this.numSplits`. In fact, it mostly depends on `this.numSplits` (aka `k`). But for the last split, there might be fewer leaves than `this.numSplits + 1`.